email-reader/src/test/java/no/sikt/nva/email/reader/util/FakeS3ClientThrowingExceptionWhenInsertingZipFile.java

```diff
@@ -0,0 +1,26 @@
+package no.sikt.nva.email.reader.util;
+
+import no.unit.nva.stubs.FakeS3Client;
+import software.amazon.awssdk.core.sync.RequestBody;
+import software.amazon.awssdk.services.s3.model.PutObjectRequest;
+import software.amazon.awssdk.services.s3.model.PutObjectResponse;
+
+
+public class FakeS3ClientThrowingExceptionWhenInsertingZipFile extends FakeS3Client {
+    private boolean firstPutObjectRequest = true;
+
+
+    @Override
+    public PutObjectResponse putObject(PutObjectRequest putObjectRequest, RequestBody requestBody) {
+        return allowInsertS3Event(putObjectRequest, requestBody);
+    }
+
+    private PutObjectResponse allowInsertS3Event(PutObjectRequest putObjectRequest, RequestBody requestBody) {
+        if (firstPutObjectRequest) {
+            firstPutObjectRequest = false;
```

Comment:
why do you do this? is there something special needing another call to s3 client?

Reply:
The first call is to insert the s3 event that triggers the lambda. And the following are the lambda trying to add the zip-files. But yeah, not a design award. Couldn't figure out a rule that would fail all the persisting of the zip-files but allow the first event to get passed.